### PR TITLE
fix: improve fractional scale

### PR DESCRIPTION
**Pull request by lychee-yycy**

### CODE REVIEW

File: src/types/qwcompositor.cpp [hunk 1]
```diff
@@ -8,6 +8,9 @@
 #include "qwoutput.h"
 #include "util/qwsignalconnector.h"
 #include "render/qwrenderer.h"
+#if WLR_VERSION_MINOR > 16
```
Comment (wineee) [inline src/types/qwcompositor.cpp:11]: only support 0.17/0.18 now

**Review by wineee — CHANGES_REQUESTED** [review]

File: src/types/qwcompositor.cpp (hunk 1 above)
Comment (lychee-yycy) [reply src/types/qwcompositor.cpp:11]: ref:https://gitlab.freedesktop.org/wlroots/wlroots/-/releases 
key word: wp-fractional-scale-v1

Comment (wineee) [reply src/types/qwcompositor.cpp:11]: qwlroots do not support WLR_VERSION_MINOR <= 16,  just delete this line

File: src/types/qwcompositor.cpp [hunk 2]
```diff
@@ -298,6 +299,11 @@ void QWSurface::unlockCached(uint32_t seq)
     wlr_surface_unlock_cached(handle(), seq);
 }
 
+void QWSurface::setFractionalScale(double scale)
+{
+    QWFractionalScaleManagerV1::notifyScale(this, scale);
```
Comment (wineee) [inline src/types/qwcompositor.cpp:304]: @zccrs

Comment (zccrs) [reply src/types/qwcompositor.cpp:304]: NO, Don't add this function. You can direct use QWFractionalScaleManagerV1::notifyScale in compositor.

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**